CMakeLists.txt

```diff
@@ -174,4 +174,6 @@ add_subdirectory(source)
 add_subdirectory(tools)
 add_subdirectory(test)
 
-FetchContent_MakeAvailable(freertos_kernel cmock)
+if(FREERTOS_PLUS_TCP_FETCH_FREERTOS)
+    FetchContent_MakeAvailable(freertos_kernel cmock)
+endif()
```

Comment:
CMake documentation (for the latest release) mentions
>True if the constant is 1, ON, YES, TRUE, Y, or a non-zero number (including floating point numbers). False if the constant is 0, OFF, NO, FALSE, N, IGNORE, NOTFOUND, the empty string, or ends in the suffix -NOTFOUND.

We should keep default behavior to clone the FreeRTOS kernel. This default behavior is used in [this now failing check](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/actions/runs/9763074732/job/26950437825?pr=1166). Maybe negate your variable to something like `FREERTOS_PLUS_TCP_USE_LOCAL_FREERTOS`. Then your condition could be...

```
if(NOT FREERTOS_PLUS_TCP_USE_LOCAL_FREERTOS)
    FetchContent_MakeAvailable(freertos_kernel cmock)
endif()
```

Reply:
Great input, indeed this breaks existing builds. Thanks for the pointer.

I flipped the logic, hopefully CI passes now.

Reply:
Perhaps it makes sense to check if we are the top level CMake project and only clone FreeRTOS kernel then?

Reply:
I didn't know this was possible until I googled it just now. Is doing [this](https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html) considered best practice?

Reply:
@archigup I've changed the logic as you suggested